### ISSUE TITLE
[USER ISSUE] Multiple aliases pointing to the same bundle spec are not supported

### ISSUE THREAD

**Issue by hao-fang** [codalab/codalab-worksheets#2021]
**Describe the bug**
When I set two bundle dependencies with the same bundle spec but different aliases, one of them is get dropped.

**Codalab Information**
- *Your username*: hfang
- *UUID of worksheet where problem occurred*: N/A
- *UUID of problem bundle (please don't delete)*: 0x218c2d796c9847a283817362459eb308
- *Error message received from Codalab (if applicable)*: N/A
 - *OS (if on CLI)*: N/A
 - *Codalab CLI Version (if on CLI)*:  N/A
 - *Browser (if on Web)*; Chrome

**To Reproduce**
Steps to reproduce the behavior :
```
run foo:0xea12e8bbfa9d4e0788dfef6457df7cee/foo foo2:0xea12e8bbfa9d4e0788dfef6457df7cee --- "cat foo2"
```

**Expected behavior**
Both "foo" and "foo2" should be accessible and shown as dependencies.

**Screenshots**
N/A

**Additional context**
This is a sample job to illustrate the problem.
It happens for a complicated worksheet in practice, when sometimes the two dependencies can be different, but sometimes they point to the same bundle spec.

**Comment by candicegjing** [comment]
@hao-fang Just want to double-check with you that in the example you provided above:
<img width="663" alt="Screen Shot 2020-03-23 at 4 40 22 PM" src="https://user-images.githubusercontent.com/1483372/77373276-0d257e80-6d25-11ea-8014-6dff83c64c1d.png">
The bundle specs appear to be different in `foo` and `foo2`. Do you actually mean the following case where both `foo` and `foo2` are pointing to `0xea12e8bbfa9d4e0788dfef6457df7cee`?
```
cl run foo:0xea12e8bbfa9d4e0788dfef6457df7cee foo2:0xea12e8bbfa9d4e0788dfef6457df7cee --- "cat foo2"
```

**Comment by hao-fang** [comment]
@candicegjing no, I intentionally make them different in terms of the subpaths, while their bundle IDs are the same.
But I just tried `cl run foo:0xea12e8bbfa9d4e0788dfef6457df7cee foo2:0xea12e8bbfa9d4e0788dfef6457df7cee --- "cat foo2"`, and it doesn't work either.

**Comment by nelson-liu** [comment]
I don't think this is working---not sure if it is a regression, but here's an example bundle: https://worksheets.codalab.org/bundles/0x2488d858022e4d268708e5839bd4afde . I mount a bundle (`LICENSE.txt`, just a sample bundle here) at `a` and `b`. however, it seems like it only exists at `b`, and `a` is a broken symlink.

**Comment by adiprerepa** [comment]
was able to repro, looking into it. it seems that the first dependency declared in `cl run` seems to be the broken one.

**Comment by epicfaace** [comment]
@adiprerepa how is work on this issue going? Would you prefer if someone else picks this up?

**Comment by adiprerepa** [comment]
yeah, that would be fine
